src/pages/todo/components/TodoForm.js

```diff
@@ -0,0 +1,136 @@
+import React from 'react'
+import {reduxForm} from 'redux-form'
+import moment from 'moment'
+import DayPicker, { DateUtils } from 'react-day-picker'
+import { CompactPicker } from 'react-color';
+import { UPDATE_TODO } from '../actions'
+
+// Library styling
+if (process.env.BROWSER) {
+  require("react-day-picker/lib/style.css")
+}
+
+const fields = [
+  'id', 
+  'title', 
+  'due_date', 
+  'color', 
+  'marked', 
+]
+
+const validate = values => {
+  const errors = {}
+  if (values.title == "") {
+    errors.title = 'Required'
+  }
+  return errors
+}
+
+class TodoForm extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {editing: false}
+  }
+
+  _handleCheckbox(event) {
+    let { fields: {marked} } = this.props
+    let data = event.target.checked ? 1 : 0
+    marked.onChange(data)
+  }
+
+  _handleDay(event, data) {
+    let { fields: {due_date} } = this.props
+    due_date.onChange(data.getTime()/1000)
+  }
+
+  _handleColor(data) {
+    let { fields: {color} } = this.props
+    color.onChange(data.hex)
+  }
+
+  _showForm() {
+    this.setState({editing : true})
+  }
+
+  _handleSubmit(data) {
+    if(data.marked) {
+      data['marked'] = 1
+    } else {
+      data['marked'] = 0
+    }
+    this.props.dispatch({type: UPDATE_TODO.REQUEST, data: data, id: data.id})
+    this.setState({editing : false})
```

Comment:
Consider async update for editing as in TodoListItem; use fetchState to do this.